Makefile

```diff
@@ -12,138 +12,258 @@
+	$(GOLANGCI_LINT) run -v --fast=false
+
+lint-charts: $(HELM) # Run helm lint tests
+	helm lint --strict charts/secrets-store-csi-driver
+	helm lint --strict manifest_staging/charts/secrets-store-csi-driver
+
+## --------------------------------------
+## Builds
+## --------------------------------------
+.PHONY: build
+build:
+	CGO_ENABLED=0 GOOS=linux go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+
+.PHONY: build-windows
+build-windows:
+	CGO_ENABLED=0 GOOS=windows go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+
+.PHONY: build-darwin
+build-darwin:
+	CGO_ENABLED=0 GOOS=darwin go build -a -ldflags $(LDFLAGS) -o _output/secrets-store-csi ./cmd/secrets-store-csi-driver
+
+.PHONY: container
+container:
+	docker build --no-cache --build-arg LDFLAGS=$(LDFLAGS) -t $(IMAGE_TAG) -f docker/Dockerfile .
+
+.PHONY: container-linux
+container-linux: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="linux/$(ARCH)" --build-arg LDFLAGS=$(LDFLAGS) \
+ 		-t $(IMAGE_TAG)-linux-$(ARCH) -f docker/Dockerfile .
+
+.PHONY: container-windows
+container-windows: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="windows/$(ARCH)" --build-arg LDFLAGS=$(LDFLAGS) \
+		--build-arg BASEIMAGE=mcr.microsoft.com/windows/nanoserver:$(OSVERSION) \
+		--build-arg BASEIMAGE_CORE=mcr.microsoft.com/windows/servercore:$(OSVERSION) \
+ 		-t $(IMAGE_TAG)-windows-$(OSVERSION)-$(ARCH) -f docker/windows.Dockerfile .
+
+.PHONY: docker-buildx-builder
+docker-buildx-builder:
+	@if ! docker buildx ls | grep -q container-builder; then\
+		DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name container-builder --use --driver-opt image=moby/buildkit:$(BUILDKIT_VERSION);\
+	fi
+
+.PHONY: container-all
+container-all:
+	$(MAKE) container-linux
+	for osversion in $(ALL_OSVERSIONS.windows); do \
+  		OSVERSION=$${osversion} $(MAKE) container-windows; \
+  	done
+
+.PHONY: push-manifest
+push-manifest:
+	docker manifest create --amend $(IMAGE_TAG) $(foreach osarch, $(ALL_OS_ARCH), $(IMAGE_TAG)-${osarch})
+	# add "os.version" field to windows images (based on https://github.com/kubernetes/kubernetes/blob/master/build/pause/Makefile)
+	set -x; \
+	registry_prefix=$(shell (echo ${REGISTRY} | grep -Eq ".*[\/\.].*") && echo "" || echo "docker.io/"); \
+	manifest_image_folder=`echo "$${registry_prefix}${IMAGE_TAG}" | sed "s|/|_|g" | sed "s/:/-/"`; \
+	for arch in $(ALL_ARCH.windows); do \
+		for osversion in $(ALL_OSVERSIONS.windows); do \
+			BASEIMAGE=mcr.microsoft.com/windows/nanoserver:$${osversion}; \
+			full_version=`docker manifest inspect $${BASEIMAGE} | jq -r '.manifests[0].platform["os.version"]'`; \
+			sed -i -r "s/(\"os\"\:\"windows\")/\0,\"os.version\":\"$${full_version}\"/" "${HOME}/.docker/manifests/$${manifest_image_folder}/$${manifest_image_folder}-windows-$${osversion}-$${arch}"; \
```

Comment:
what version of Docker runs in actions? can we replace this with `docker manifest annotate`?

Reply:
cloudbuild uses docker version `19.03.5` and `--os-version` isn't supported in that with docker manifest annotate. We can update to using `docker manifest annotate` once the docker version is updated.